R/geneset_activity.R

```diff
@@ -0,0 +1,106 @@
+#' Calculating gene set activity inside signatures
+#'
+#' Calculates activity of signatures only using genes in the provided gene sets
+#'
+#' @param signature_geneset_df a data.frame with the first column specifying
+#' signature names and second column specifying gene set names.
+#' @param gene_set_list a named list storing all possible gene sets with each
+#' element being a vector of gene IDs. It is used to extract genes IDs given
+#' the name of a gene set.
+#' @param input_data a data.frame with gene IDs in the first column and
+#' expression values from the second column.
+#' @param model the ADAGE model to be used for calculating signature activity.
+#' @param HW_cutoff number of standard deviations away from mean in a node's
+#' weight distribution to be considered as high-weight (default to 2.5).
+#' @return a data.frame with the first column being "signature|geneset" names and
+#' the rest columns storing activities for every sample in the input_data.
+#' @export
+signature_geneset_activity <- function(signature_geneset_df, gene_set_list,
+                                       model, input_data, HW_cutoff = 2.5){
+
+  if (!check_input(input_data)){
+    stop("The input data should be a data.frame with the first column as a
+         character of gene IDs and the rest of the columns storing numeric
+         expression values for each sample.")
+  }
+
+  if (!check_input(model)) {
+    stop("The model should be a data.frame with the first column as a character
+         of gene IDs and the rest of the columns storing numeric weight values
+         for each node.")
+  }
+
+  if(!all(input_data[, 1] == model[, 1])){
+    stop("The gene IDs stored in the first column of the input data and the
+         first column of the model should be the same.")
+  }
+
+  weight_matrix <- as.matrix(model[, -1])
+  rownames(weight_matrix) <- model[[1]]
```

Comment:
`model` is a `tibble`. You use `tibble[[index]]` to get a column vector. `tibble[, index]` would return anther `tibble`. It is a little strange and needs to be careful.